### PR TITLE
feat(inode): add inode struct

### PR DESCRIPTION
**Pull request by overbool**
Closes: https://github.com/ipfs/go-mfs/issues/11

Closes: https://github.com/ipfs/go-ipfs/pull/5182
Closes: https://github.com/ipfs/go-ipfs/issues/5094

### CODE REVIEW

**Comment by overbool** [comment]
@schomatis could u help me review it?

**Comment by schomatis** [comment]
This looks good, as part of removing the inode interface I would also remove the accessor functions (`DagService`, `Parent`, `Name`) since they offer no additional logic (as was suggested to me in the original PR) allowing to access the inode attributes like `file.DagService` (instead of calling `file.DagService()`); moving also the function comments to the attributes in the structure.

**Comment by overbool** [comment]
@schomatis Yes, It's really unnecessary to call function. I had fixed it according to your suggestion.

File: inode.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import (
+	ipld "github.com/ipfs/go-ipld-format"
+)
+
+// inode abstracts the common characteristics of the MFS `File`
+// and `Directory`. All of its attributes are initialized at
+// creation.
+type inode struct {
+	name       string
```
Comment (schomatis) [inline inode.go]: Could you please put here the comments that were in the removed accessor functions? That way we could leverage them to document this structure.

Comment (overbool) [reply inode.go]: OK

Comment (overbool) [reply inode.go]: @schomatis it's ok, I had added the comments here.

**Pull request merged**